esp-hal-embassy/src/executor/interrupt.rs

```diff
@@ -4,7 +4,7 @@ use core::{cell::UnsafeCell, mem::MaybeUninit};
 
 use embassy_executor::{raw, SendSpawner};
 use esp_hal::{
-    get_core,
+    core,
```

Comment:
Just in this file, or in general?

Reply:
In general I think, I don't know if Rust can cope with a function `core` and the `core` crate together. Though I guess if it can, we can leave it and come back later.

Reply:
Okay will do so in another PR, this one is a bit of a beast and I don't really want to touch it if CI is happy and I don't need to 😅

Reply:
Let's make `get_core` private and use `Cpu::current()` everywhere (though we don't have to do that here if we can get away with it).